configure

```diff
@@ -83,6 +83,61 @@ cxx_works()
     fi
 }
 
+# Test if the compiler successfully compiles the specified source file. If the optional compilation
+# argument is specified ($3) then pass it to the compiler along with the already-established
+# arguments (this can be used to test if the compiler accepts the argument).
+#   $1 - the name of the source file to compile
+#   $2 - the name of the result object file
+#   $3 - the argument to test (optional)
+#   CXX - the name of the compiler
+#   CXXFLAGS, CXXFLAGS_EXTRA - any predetermined flags
+try_compile_source()
+{
+    eval $CXX $CXXFLAGS ${3:-} $CXXFLAGS_EXTRA '"$configtmpdir"/"$1"' \
+            -c -o '"$configtmpdir"/"$2"' > /dev/null 2>&1
+}
+
+# Test if the linker successfully links the specified object file. If the optional link argument
+# is specified ($3) then pass it to the linker along with the already-established arguments
+# (this can be used to test if the linker accepts the argument).
+#   $1 - the name of the object file to link
+#   $2 - the name of the result executable file
+#   $3 - the argument to test (optional)
+#   CXX - the name of the compiler
+#   LDFLAGS, LDFLAGS_EXTRA - any predetermined flag
+try_link_object()
+{
+    if [ ! -e "$configtmpdir"/"$1" ]; then
+        # If the object file doesn't exist yet, need to compile
```

Comment:
This behaviour isn't documented, why is testing whether the object already exists?

Reply:
This is not documented and also it's not really necessary, I mean, instead of guessing source file name when object file doesn't exist just error out and make it caller responsibility to make sure object file exists.